build.gradle

```diff
@@ -81,6 +81,7 @@ if (JavaVersion.current() == JavaVersion.VERSION_1_8) {
                               "--add-opens", "java.base/jdk.internal.reflect=ALL-UNNAMED",
                               "--add-opens", "java.sql/java.sql=ALL-UNNAMED",
                               "--add-opens", "java.base/sun.util.calendar=ALL-UNNAMED",
+                              "--add-opens", "java.base/sun.net.util=ALL-UNNAMED",
```

Comment:
can you please describe why this is needed? Which test required this?

Reply:
CI also runs the tests with JDK17 so I'm surprised that this would fail with JDK17. Could you please double-check whether tests pass with JDk17 for you without this change?

Reply:
It was the `TestIcebergSourceWithWatermarkExtractor` which is reverted now. Pls see more details down below at the answer to your comment regarding `TestIcebergSourceWithWatermarkExtractor`.